### PR TITLE
[LoRA] enable LoRA for Mochi-1

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Needed for fine-tuning.

### CODE REVIEW

File: src/diffusers/loaders/lora_pipeline.py
Comment (sayakpaul) [inline src/diffusers/loaders/lora_pipeline.py:2365]: Unrelated change but doesn't hurt I guess.

File: src/diffusers/loaders/lora_pipeline.py
Comment (sayakpaul) [inline src/diffusers/loaders/lora_pipeline.py:2670]: A copy-paste of the Cog LoRA loader classes, indicated by the "Copied from ..." comments.

File: src/diffusers/loaders/lora_pipeline.py
Comment (a-r-r-o-w) [reply src/diffusers/loaders/lora_pipeline.py:2365]: Looks okay to fix here

File: tests/lora/test_lora_layers_mochi.py
Comment (a-r-r-o-w) [inline tests/lora/test_lora_layers_mochi.py:143]: Looks good since not supporting T5 finetuning!

**Review by a-r-r-o-w — APPROVED** [review]
Looks great! Don't see any major differences in the lora mixin, so I think everything should be good to merge once the finetuning script is working - going to take a look at it now, thanks!

**Comment by sayakpaul** [comment]
> so I think everything should be good to merge once the finetuning script is working

@a-r-r-o-w thanks! Do you think it could make sense to merge regardless with the tests, etc.? Because this will also allow others to experiment from `main`. I am okay otherwise too.

**Review by yiyixuxu — APPROVED** [review]

**Comment by sayakpaul** [comment]
The tests failing are related to fact that our CI is currently in PyTorch 2.4. https://github.com/huggingface/diffusers/pull/9961 should fix this. Additional info:
https://huggingface.slack.com/archives/C014N4749J9/p1732002990541779

**Comment by sayakpaul** [comment]
Seems like the `NaN` tests are failing with PyTorch 2.5 and when using "CPU" as the test device. Will investigate deeper. It doesn't happen on PyTorch 2.4. Note that all the tests pass when using a GPU even with PyTorch 2.5. 

@yiyixuxu can I merge this PR or should we first investigate the `NaN` issue? It's failing for other PRs too. 

Cc: @BenjaminBossan as well. It's the `safe_merge` argument in the `merge()` method of `peft`. Does this sound familiar?

**Comment by yiyixuxu** [comment]
> @yiyixuxu can I merge this PR or should we first investigate the NaN issue? It's failing for other PRs too.

 @a-r-r-o-w wants to test it with fine-tune script first, no?

**Comment by sayakpaul** [comment]
> @a-r-r-o-w wants to test it with fine-tune script first, no?

Well, that is currently done by me here https://github.com/a-r-r-o-w/cogvideox-factory/pull/90/ with Aryan's reviews. However, I don't think the LoRA implementation is dependent on the fine-tuning experiments as it's less likely gonna change (and it's similar to CogVideoX). But okay to wait and will defer to @a-r-r-o-w.

**Comment by a-r-r-o-w** [comment]
I think should be okay to merge without waiting for finetuning script to work or having an available checkpoint since it will unblock others trying to work on finetuning scripts based on Diffusers (I don't know if there is anyone apart from us yet though). Thanks!

**Pull request merged**
